app/models/lookup_value.rb

```diff
@@ -28,7 +28,7 @@ class LookupValue < ActiveRecord::Base
   scoped_search :in => :lookup_key, :on => :key, :rename => :lookup_key, :complete_value => true
 
   def value_present?
-    self.errors.add(:value, :blank) if value.nil? && !use_puppet_default
+    self.errors.add(:value, :blank) if value.to_s.empty? && !use_puppet_default && lookup_key.puppet?
```

Comment:
This change to `.to_s.empty?` is to prevent 'false' values being recognized as `value.nil?` right? 

The 2nd part of the conditional feels a bit like feature envy, maybe you would want to move the LookupValue puppet logic to a subclass (doesn't need to be an ActiveRecord model) and the default puppet lookup value to another class. Or do it through composition, perhaps it's simpler, I didn't go that route because it seemed easy to do this way.

Something like this:

```ruby
  class PuppetLookupValue < LookupValue
    attr_accessible :use_puppet_default
    validate :puppet_lookup_key?
    validate :value_present?

    private

    def puppet_lookup_key?
      lookup_key.puppet?
    end

    def value_present?
      errors.add(:value, :blank) if value.nil?
    end
  end

  class PuppetDefaultLookupValue < PuppetLookupValue
    before_validation :validate_and_cast_value
  end
```